test/e2e/controllerid_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2020 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"testing"
+)
+
+func testControllerCorrectIDPrometheusServer(t *testing.T) {
+	t.Parallel()
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+	framework.SetupPrometheusRBAC(context.Background(), t, testCtx, ns)
+
+	name := "test"
+	p := framework.MakeBasicPrometheus(ns, name, name, 1)
+	p.Annotations["operator.prometheus.io/controller-id"] = "88"
+
+	_, err := framework.CreatePrometheusAndWaitUntilReady(context.Background(), ns, p)
+	if err == nil {
```

Comment:
Each test will have to wait for 5m which would make the tests duration longer than needed.
If we have a successful test running 2 operators in parallel managing separate objects (e.g. "testControllerMultipleOperators"), it will assert the same behavior.

Reply:
right